js/src/DappRequests/store.js

```diff
@@ -41,131 +38,137 @@ export default class Store {
+      accumulator[appId][methodGroup].push(request);
+      return accumulator;
+    }, {});
   }
 
-  @action createToken = (appId) => {
+  @action createToken = appId => {
     const token = sha3(`${appId}:${Date.now()}`);
 
     this.tokens = Object.assign({}, this.tokens, {
       [token]: appId
     });
 
     return token;
-  }
-
-  @action removeRequest = (_queueId) => {
-    this.requests = this.requests.filter(({ queueId }) => queueId !== _queueId);
-    delete this.sources[_queueId];
-  }
-
-  @action queueRequest = (request) => {
-    const { data, origin, source } = request;
-    const appId = this.tokens[data.token];
-
-    let queueId = ++nextQueueId;
+  };
 
-    this.sources[queueId] = source;
-    this.requests = this.requests.concat([{
-      appId,
-      queueId,
-      request: {
-        data,
-        origin
-      }
-    }]);
-  }
+  @action queueRequest = request => {
+    const { data: { method, token } } = request;
+    const appId = this.tokens[token];
 
-  @action addTokenPermission = (method, token) => {
-    const id = `${method}:${this.tokens[token]}`;
-
-    this.permissions = Object.assign({}, this.permissions, {
-      [id]: true
-    });
-    this.savePermissions();
-  }
+    // Add a new request in this.requests
+    this.requests = {
+      ...this.requests,
+      [`${method}:${appId}`]: request
+    };
+  };
 
-  @action approveSingleRequest = ({ queueId, request: { data } }) => {
-    const source = this.sources[queueId];
+  @action approveRequest = (method, appId) => {
+    const { source, data } = this.requests[`${method}:${appId}`];
 
-    this.removeRequest(queueId);
+    this.addAppPermission(method, appId);
+    this.removeRequest(method, appId);
 
     if (data.api) {
       this.executePubsubCall(data, source);
     } else {
       this.executeMethodCall(data, source);
     }
-  }
+  };
 
-  @action approveRequest = (queueId, approveAll) => {
-    const queued = this.findRequest(queueId);
+  @action approveRequestGroup = (groupId, appId) => {
+    // Get methods of this requestGroup
+    const { methods } = methodGroups[groupId];
 
-    if (approveAll) {
-      const { request: { data: { method, token, params } } } = queued;
+    methods.forEach(method => {
+      if (!this.requests[`${method}:${appId}`]) {
+        return;
+      }
 
-      this.getFilteredSection(method || params[0]).methods.forEach((m) => {
-        this.addTokenPermission(m, token);
-        this.findMatchingRequests(m, token).forEach(this.approveSingleRequest);
-      });
-    } else {
-      this.approveSingleRequest(queued);
-    }
-  }
+      this.approveRequest(method, appId);
+    });
+  };
 
-  @action rejectRequest = (queueId) => {
-    const { request: { data } } = this.findRequest(queueId);
-    const source = this.sources[queueId];
+  @action rejectRequest = (method, appId) => {
+    const { source, data } = this.requests[`${method}:${appId}`];
```

Comment:
`${method}:${appId}` is scattered around in many places, maybe put it in a helper method?

Reply:
Fixed